libcontainer/nsenter/nsexec.c

```diff
@@ -192,22 +194,74 @@ static void update_setgroups(int pid, enum policy_t setgroup)
 	}
 }
 
-static void update_uidmap(int pid, char *map, size_t map_len)
+static int attempt_mapping_tool(const char *app, int pid, char *map, size_t map_len)
+{
+#define MAX_ARGV 20
+	char *argv[MAX_ARGV];
+	char pid_fmt[16];
+	int argc = 0;
+	size_t i;
+	int child;
+
+	sprintf (pid_fmt, "%d", pid);
```

Comment:
It's just because I don't like that you've hard-coded the length of `pid_fmt` to be `11`.

Personally I like @teddyking's approach to the formatting a bit more (we send most of the information from Go with a netlink message rather than formatting everything in C). I prefer that because the context in which this C code is run is quite fragile and I get less worried if we do things in Go when possible. I'll read through his PR again and point out which bits seem good to me.